File: api/src/main/java/com/salesianos/triana/ArtApi/dto/Publicacion/CreatePublicationDTO.java
package com.salesianos.triana.ArtApi.dto.Publicacion;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

public record CreatePublicationDTO(

        @NotBlank(message = "El título es obligatorio")
        @Size(min = 3, max = 31, message = "El título debe tener entre 3 y 31 caracteres")
        String titulo,

        @NotBlank(message = "La descripcion es obligatoria")
        String descripcion,

        @NotBlank(message = "El tamanyo de las dimensiones es obligatorio")
        String tamanyoDimensiones,

        @NotBlank(message = "La direccion de la obra es obligatoria")
        String direccionObra,

        @NotBlank(message = "El nombre del museo es obligatorio")
        String nombreMuseo,

        @NotBlank(message = "La latitud es obligatoria")
        @Pattern(regexp = "-?\\d+(\\.\\d+)?", message = "La latitud debe tener un formato válido")
        String lat,

        @NotBlank(message = "La longitud es obligatoria")
        @Pattern(regexp = "-?\\d+(\\.\\d+)?", message = "La longitud debe tener un formato válido")
        String lon,

        String image,

        @NotNull(message = "El número de categoría es obligatorio")
        Long numeroCategoria
) {}

File: mobile/lib/models/dto/publication/create_publication_dto.dart
class CreatePublicationDTO {
  String? titulo;
  String? descripcion;
  String? tamanyoDimensiones;
  String? direccionObra;
  String? nombreMuseo;
  String? lat;
  String? lon;
  String? image;
  int? numeroCategoria;

  CreatePublicationDTO(
      {this.titulo,
      this.descripcion,
      this.tamanyoDimensiones,
      this.direccionObra,
      this.nombreMuseo,
      this.lat,
      this.lon,
      this.image,
      this.numeroCategoria});

  CreatePublicationDTO.fromJson(Map<String, dynamic> json) {
    titulo = json['titulo'];
    descripcion = json['descripcion'];
    tamanyoDimensiones = json['tamanyoDimensiones'];
    direccionObra = json['direccionObra'];
    nombreMuseo = json['nombreMuseo'];
    lat = json['lat'];
    lon = json['lon'];
    image = json['image'];
    numeroCategoria = json['numeroCategoria'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['titulo'] = this.titulo;
    data['descripcion'] = this.descripcion;
    data['tamanyoDimensiones'] = this.tamanyoDimensiones;
    data['direccionObra'] = this.direccionObra;
    data['nombreMuseo'] = this.nombreMuseo;
    data['lat'] = this.lat;
    data['lon'] = this.lon;
    data['image'] = this.image;
    data['numeroCategoria'] = this.numeroCategoria;
    return data;
  }
}
